pyxem/signals/diffraction2d.py

```diff
@@ -1115,6 +1117,50 @@ def template_match_ring(self, r_inner=5, r_outer=7, inplace=False, **kwargs):
             normalize_template_match, template=ring, inplace=inplace, **kwargs
         )
 
+    def filter_signal_axes(self, method="gaussian_filter", **kwargs):
+        if isinstance(method, str):
+            name = "scipy.ndimage"
+            _method = getattr(import_module(name), method)
+        return self.map(_method, **kwargs)
+
+    def get_num_chunked_axes(self):
+        axes = np.arange(self.data.ndim)
+        unspanned_dim = set(axes) - set(self.spanned_dimensions())
+        chunked_dim = len(unspanned_dim)
+        return chunked_dim
+
+    def filter(self, method="gaussoan_filter", inplace=False, **kwargs):
```

Comment:
The name is a bit misleading because looking at https://image.dask.org/en/latest/coverage.html, it can also be used for functions, which are not "filters", for example, morphological operation. I don't have a better name to suggest though! :)

Regarding the `method` argument, I think that it would be more transparent (no ambiguity with what function/implementation is being used) to the user and more in line with the `map` method if the argument was taking a python function instead of a string.

Reply:
Yea I think you are right!  We can just have a function `filter` which operates on the entire dataset.